flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/ZooKeeperLeaderElectionConnectionLossTest.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.leaderelection;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.HighAvailabilityOptions;
+import org.apache.flink.core.testutils.OneShotLatch;
+import org.apache.flink.runtime.util.ZooKeeperUtils;
+import org.apache.flink.runtime.zookeeper.ZooKeeperResource;
+import org.apache.flink.util.TestLogger;
+
+import org.apache.flink.shaded.curator4.org.apache.curator.framework.CuratorFramework;
+import org.apache.flink.shaded.curator4.org.apache.curator.framework.state.ConnectionState;
+import org.apache.flink.shaded.curator4.org.apache.curator.framework.state.ConnectionStateListener;
+
+import org.apache.zookeeper.KeeperException.ConnectionLossException;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.fail;
+
+/** Test behaviors of {@link ZooKeeperLeaderElectionDriver} on {@link ConnectionLossException} */
+public class ZooKeeperLeaderElectionConnectionLossTest extends TestLogger {
+
+    private static final String LATCH_PATH = "/latch";
+    private static final String LEADER_PATH = "/leader";
+
+    private static final Duration TIMEOUT = Duration.ofMillis(2000L);
+
+    @Rule public final ZooKeeperResource zooKeeperResource = new ZooKeeperResource();
+
+    @Test
+    public void testKeepLeadershipOnConnectionLoss() throws Exception {
```

Comment:
Following @mxm 's naming suggestion, maybe `testKeepLeadershipOnConnectionSuspended` would be more precise. Similar for `connectionLossLatch` -> `connectionSuspendedLatch` below